src/main/java/com/amazonaws/secretsmanager/caching/cache/SecretCacheObject.java

```diff
@@ -281,24 +237,17 @@ public boolean refreshNow() throws InterruptedException {
      *
      * @return The cached GetSecretValue result.
      */
-    public GetSecretValueResult getSecretValue() {
+    public GetSecretValueResponse getSecretValue() {
         synchronized (lock) {
             refresh();
             if (null == this.data) {
-                if (null != this.exception) { throw this.exception; }
+                if (null != this.exception) { throw new RuntimeException(this.exception); }
             }
-            GetSecretValueResult gsv = this.getSecretValue(this.getResult());
+            GetSecretValueResponse gsv = this.getSecretValue(this.getResult());
 
             // If there is no cached result, return null.
             if (null == gsv) { return null; }
 
-            // We want to clone the result to prevent callers from modifying
-            // the cached data.
-            gsv = gsv.clone();
```

Comment:
It looks like the new sdk response object tries to maintain immutability of the response.  This is great but I am not certain it will always be guaranteed.  Additionally, I am not certain this will be true for all elements within the response.

To error on the side of safety, it would be best to keep the clone logic.  What options do we have to keep this functionality?